src/freenet/node/Node.java

```diff
@@ -1603,6 +1603,12 @@ public Long get() {
 			@Override
 			public void set(Long amountOfDataToCheckCompressionRatio) {
 				synchronized(Node.this) {
+					if (amountOfDataToCheckCompressionRatio < 0 || amountOfDataToCheckCompressionRatio > 100 * 1024 * 1024) {
```

Comment:
How do we handle the `0` case? Does the compression test stuff do something sensible here?

Reply:
If I am reading this correctly, it just will never run checkCompressionEffect be cause the ratio would be 0 and the ++i  jumps past that.

	int bufferSize = 32768;
			byte[] buffer = new byte[bufferSize];
			long iterationToCheckCompressionRatio = amountOfDataToCheckCompressionRatio / bufferSize;
			int i = 0;
			while(true) {
				int l = (int) Math.min(buffer.length, maxReadLength - read);
				int x = l == 0 ? -1 : is.read(buffer, 0, l);
				if(x <= -1) break;
				if(x == 0) throw new IOException("Returned zero from read()");
				gos.write(buffer, 0, x);
				read += x;
				if(cos.written() > maxWriteLength)
					throw new CompressionOutputSizeException();

				if (++i == iterationToCheckCompressionRatio && minimumCompressionPercentage != 0) {
					checkCompressionEffect(read, cos.written(), minimumCompressionPercentage);
				}

Reply:
Heh, only for small values of “never,” though… with a large enough file `++i` can overflow. Granted, that will only happen after 140 TiB but you never know. 😁